### PR TITLE
feat: add change stream support

### LINKED ISSUE #228: Add support for CHANGE STREAM DDL
https://cloud.google.com/spanner/docs/change-streams/manage

### PR DESCRIPTION
**Pull request by taka-oyama**
Closes #228 

## Summary by CodeRabbit

- **New Features**
	- Introduced methods for creating, altering, and dropping change streams, enhancing database management capabilities.
	- Added a new class for defining change stream configurations, allowing for detailed customization of data capture.
	- Implemented an enumeration to specify different types of change stream value captures.
	- Added support for snapshot queries and improved query handling with new methods.
	- Expanded documentation on Change Streams, Secondary Index Options, and Mutations for enhanced database interactions.

- **Tests**
	- Added comprehensive tests to ensure the functionality of change streams, covering creation, alteration, and deletion scenarios.

### CODE REVIEW

**Comment by taka-oyama** [comment]
test now passing

File: tests/Schema/BlueprintTest.php [hunk 1]
```diff
@@ -513,6 +514,151 @@ public function test_dropSequenceIfExists(): void
...
+        $blueprint->uuid('id')->primary();
+        $blueprint->create();
+        $blueprint->createChangeStream($streamName)
+                ->for($blueprint->getTable())
+                ->excludeTtlDeletes(true)
+                ->excludeInsert(true)
+                ->excludeUpdate(true)
+                ->excludeDelete(true)
+                ->valueCaptureType(ChangeStreamValueCaptureType::NewRow)
+                ->retentionPeriod('1d');
```
Comment (Uyan712) [inline tests/Schema/BlueprintTest.php:589]:
```suggestion
            ->for($blueprint->getTable())
            ->excludeTtlDeletes(true)
            ->excludeInsert(true)
            ->excludeUpdate(true)
            ->excludeDelete(true)
            ->valueCaptureType(ChangeStreamValueCaptureType::NewRow)
            ->retentionPeriod('1d');
```

File: tests/Schema/BlueprintTest.php [hunk 2]
```diff
@@ -513,6 +514,151 @@ public function test_dropSequenceIfExists(): void
...
+        $blueprint = new Blueprint($tableName);
+        $blueprint->uuid('id')->primary();
+        $blueprint->create();
+        $blueprint->createChangeStream($streamName)->for($blueprint->getTable())->excludeTtlDeletes(true);
+        $blueprint->build($conn, $grammar);
+
+        $blueprint = new Blueprint('');
+        $blueprint->alterChangeStream($streamName)
+                ->excludeTtlDeletes(false)
+                ->retentionPeriod('7d');
```
Comment (Uyan712) [inline tests/Schema/BlueprintTest.php:627]:
```suggestion
            ->excludeTtlDeletes(false)
            ->retentionPeriod('7d');
```

**Review by Uyan712 — APPROVED** [review]
I suggested removing the unnecessary indentations, but overall, LGTM.

**Pull request merged**